EN version of the article 2025-12-06_partage_modeles_qgis (#104)
geotribu/english-blog: content/posts/2026-08-01_qgis_sharing_models.md

Reviewer: Rename file into:

`content/posts/2026-09-01_qgis_sharing_models.md`

---
title: How to share your QGIS models easily
subtitle: Five methods for easily sharing a QGIS model
authors:
  - marcducobu
categories:
  - Article
comments: true

> date: 2026-08-01

Reviewer: ```suggestion
date: 2026-09-01
```

description: "Discover 5 ways to share your QGIS models: from the paper plane method to plugin integration !"
icon: fontawesome/regular/paper-plane
image:
license: default
links:
  - Original version (French): https://geotribu.fr/articles/2025/2025-12-06_partage_modeles_qgis/
pin: false
tags:
  - QGIS
  - models
---

# Five methods for easily sharing a QGIS model

## Introduction

Finally, after long hours of work and drinking too many cups of coffee, you have created a model as beautiful as a [Kandinsky painting](https://fr.wikipedia.org/wiki/Vassily_Kandinsky#/media/Fichier:Vassily_Kandinsky,_1923_-_Circles_in_a_Circle.jpg). Now, you wish to share your model with your colleagues so you're not the only one who can enjoy it.

In this article, I will present five methods for sharing your QGIS model.

> ## The paper plane method

Reviewer: To avoid having the full article on blog homepage, you have to put an excerpt. We also usually add a button to comments div.

```suggestion
In this article, I will present five methods for sharing your QGIS model.

<!-- more -->

[Leave a comment :fontawesome-solid-comments:](#__comments "Go to comments"){: .md-button }
{: align=middle }

## The paper plane method
```

![Screenshot of the QGIS Hub extension.](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2025/partage_modeles_qgis/QGIS_HUB_PLUGIN.png){: .img-center loading=lazy }

## QGIS Resource Sharing

[QGIS Resource Sharing](https://plugins.qgis.org/plugins/qgis_resource_sharing/) is a plugin that allows you to load resources from online repositories. These repositories can contain styles, symbols, scripts and processing templates, R scripts, etc. When the plugin is installed,
you can share resources very easily.

![Screenshot of the QGIS Resource Sharing plugin.](https://qgis-contribution.github.io/QGIS-ResourceSharing/_images/repositories.png){: .img-center loading=lazy }

You can configure the plugin to use multiple sources.
Furthermore, the management of updates is automated: collections are downloaded and updated via Git, which greatly facilitates the synchronisation of shared resources.

To share your resources, you can [create your own Git repository](https://qgis-contribution.github.io/QGIS-ResourceSharing/authoring/creating-repository.html) or contribute to an existing repository.

> _Many thanks to [Julien](../../team/julien-moura.md) and [Loïc](../../team/loic-bartoletti.md) for discovering the tool!_

Reviewer 2: ```suggestion
_Many thanks to [Julien](https://geotribu.fr/team/julien-moura/) and [Loïc](https://geotribu.fr/team/loic-bartoletti/) for discovering the tool!_
```

Another site, relative links don't work here,

## Embedding your models into a plugin

> The last method is to embed your models into a QGIS plugin.
> Each plugin can add tools to the "Processing Toolbox" of QGIS, including models. For instance, the [Cadastre](https://plugins.qgis.org/plugins/cadastre/) plugin adds additional processing tools to the toolbox: exploding_head: !

Reviewer 2: ```suggestion
The last method is to embed your models into a QGIS plugin.  
Each plugin can add tools to the "Processing Toolbox" of QGIS, including models. For instance, the [Cadastre](https://plugins.qgis.org/plugins/cadastre/) plugin adds additional processing tools to the toolbox: exploding_head: !
```

If the line break is what you want, it's possible by typing two spaces at the end of the line ;-)

In this case, the [QGIS plugin Templater](https://oslandia.gitlab.io/qgis/template-qgis-plugin/) can help you in this task : it allows you to initiate a plugin structure containing a processing provider. Another tool that
can help you is the [QGIS Plugin Templater GUI](https://plugins.qgis.org/plugins/qgis_plugin_templater_gui/) which enables you to configure the
QGIS plugin Templater tool within QGIS.

But creating a plugin is not so easy, so I started developing a tool to assist you with this task. This is the [Models2Plugin](https://plugins.qgis.org/plugins/models2plugin/).

You can embed your plugin in three steps: first, encode some basic information; then, select the models to integrate; and finally, generate the plugin.

![Screenshot of Models2Plugin](https://cdn.geotribu.fr/img/articles-blog-rdp/articles/2025/partage_modeles_qgis/img_modes2plugin.png){: .img-center loading=lazy }

## To conclude

We explored five methods for sharing your QGIS models.
Each method has its own specifics and can respond to a particular need.
The QGIS community is incredibly active and creative in making the use of QGIS ever more efficient.

> Were you aware of these different methods? Do you have any other tips for sharing your models? Feel free to share your experiences in the comments!

Reviewer 2: ```suggestion
Were you aware of these different methods ? Do you have any other tips for sharing your models ? Feel free to share your experiences in the comments !
```
